Add a default to skip missing external modules during doco build

GiudGiud opened the pull request:
refs #27097

I m splitting this off from #27027 to try to get downstream apps to be green and have a smooth merge

GiudGiud committed:
Add the optional to skip external modules during doco build
refs #27097

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2285402/) on 4af2ecf wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/27933/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2285446/) on 4af2ecf wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>d7f7f7</th>
      <th colspan=4>#27933 4af2ec</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.06%</td>
      <td>85.06%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>104371</td>
      <td>104372</td>
      <td>+1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>18330</td>
      <td>18329</td>
      <td>-1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/27933/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/27933/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/stochastic_tools)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/27933/coverage/xfem)

</details>



This comment will be updated on new commits.

GiudGiud commented:
@cticenhour if I could please get a review here

cticenhour commented:
Will do so this afternoon - thanks!

cticenhour commented on python/MooseDocs/common/get_content.py:
```diff
@@ -74,8 +74,12 @@ def _doc_import(root_dir, content=None):
+        else:
+            LOG.warning("Part of the content specified in 'config.yml' is not included in documentation since " +
+                        root_dir + ' is not a valid directory')
+        return set()
```
```suggestion
        return None
```

Let's continue to return `None`, rather than an empty set. Judging from the passing tests we don't do error checking against this return (or an empty set is sufficient), but this is consistent with the other error conditions at this level of the method.

GiudGiud replied:
I really need a valid return to avoid changing the for loop afterwards

the error conditions are throwing so I dont think the for loop matters. This is moving along instead

GiudGiud replied:
umm lemme upload the changes I would have to make and you decide

GiudGiud replied:
done

cticenhour reviewed (changes requested):
Was good to split this out. Only a couple minor comments. I'd also like a unit test to hit the new warning condition.

GiudGiud committed:
Address Casey's review:
- keep a return None for the exception case of missing external module
- spelling
- add unit test

GiudGiud commented:
@cticenhour

GiudGiud commented:
awesome thanks!
